src/pianoroll/view/pianorollview.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2021 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#ifndef MU_PIANOROLL_PIANOROLLVIEW_H
+#define MU_PIANOROLL_PIANOROLLVIEW_H
+
+#include <QQuickPaintedItem>
+#include <QIcon>
+#include <QColor>
+
+#include "async/asyncable.h"
+#include "context/iglobalcontext.h"
+#include "pianoroll/ipianorollcontroller.h"
+#include "audio/iplayback.h"
+
+namespace mu::pianoroll {
+
+
+enum class NoteSelectType
+{
+    REPLACE = 0,
+    XOR,
+    ADD,
+    SUBTRACT,
+    FIRST
+};
+
+enum class DragStyle
+{
+    NONE = 0,
+    CANCELLED,
+    SELECTION_RECT,
+    NOTE_POSITION,
+    NOTE_LENGTH_START,
+    NOTE_LENGTH_END
+};
+
+struct BarPattern
+{
+    QString name;
+    char isWhiteKey[12];  //Set to 1 for white keys, 0 for black
+};
+
+struct NoteBlock
+{
+    Ms::Note* note;
+    int voice;
+    int staffIdx;
+};
+
+class PianorollView : public QQuickPaintedItem, public async::Asyncable
+{
+    Q_OBJECT
+
+public:
+    enum class PianorollTool : char { SELECT, ADD, CUT, ERASE };
+    Q_ENUM(PianorollTool)
+
+    static const BarPattern barPatterns[];
+
+private:
+    INJECT(pianoroll, context::IGlobalContext, globalContext)
+    INJECT(pianoroll, IPianorollController, controller)
+    INJECT(playback, audio::IPlayback, playback)
```

Comment:
believe it should be pianoroll for the 1st parameter here (not entirely sure what difference it makes though!)

Reply:
Fixed.

Reply:
> not entirely sure what difference it makes though!

It's actually just for information. In the future, I want to make a diagnostic page on which to show what depends on what.